src/utils.rs

```diff
@@ -389,3 +389,55 @@ impl Report {
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    #[cfg(feature = "fare-v2")]
+    mod serde_currency {
+        use super::super::*;
+        use pretty_assertions::assert_eq;
+        use serde_derive::{Deserialize, Serialize};
+
+        #[derive(Serialize, Deserialize)]
+        struct CurrencyCodeWrapper {
+            #[serde(
+                serialize_with = "ser_currency_code",
+                deserialize_with = "de_currency_code"
+            )]
+            pub currency_code: String,
+        }
+
+        #[test]
+        fn test_serde_valid_currency_code() {
+            let wrapper = CurrencyCodeWrapper {
+                currency_code: "EUR".to_string(),
+            };
+            let json = serde_json::to_string(&wrapper).unwrap();
+            let wrapper: CurrencyCodeWrapper = serde_json::from_str(&json).unwrap();
+
+            assert_eq!(wrapper.currency_code, "EUR");
+        }
+
+        #[test]
+        fn test_de_invalid_currency_code() {
+            let result: Result<CurrencyCodeWrapper, _> =
+                serde_json::from_str("{\"currency_code\":\"XXX\"}");
+            let err_msg = result.err().unwrap().to_string();
```

Comment:
`let err_msg = result.unwrap_err();`

Reply:
Indeed, I missed that one and it makes sense. Funny fact though, when I did change it, it complains because `CurrencyWrapper` does not implements `std::fmt::Debug`. The reason is because `.unwrap_err()` will panic with the value if the result is of type `Ok()`. I did the change in 62a3702f3096e9df713dc795cf81c0ddd51e2495.